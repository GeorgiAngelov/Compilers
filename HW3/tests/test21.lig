type a : [[a]];
type b : [[b]];

var x : a;
var y : b = [x];

