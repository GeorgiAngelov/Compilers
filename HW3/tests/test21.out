(decls
   (decl-type:ok (id a) (type-array (type-array (type-id a))))
   (decl-type:ok (id b) (type-array (type-array (type-id b))))
   (decl-var:ok (id x) (type-id a))
   (decl-var:NONE (id y) (type-id b) (array-lit:[a] (exps (id:a x)))))
Function main() not defined or has wrong type!
Typechecking failed!
